### DISCUSSION TITLE
access opencutlist data by Ruby

### DISCUSSION THREAD

**Discussion by tomboul26** [lairdubois/lairdubois-opencutlist-sketchup-extension#654] [General]
In my sketchup project I have a lot of components but it is not obvious to know which ones were not marked as material in opencutlist
I wanted to write a quick code in ruby, to know wich ones was not marked

I have already done a tree-browsing of all the components each in their folder, is it possible to access the opencutlist data to know if they are marked or not and how?

**Comment by bbeaulant** [comment]
Hello @tomboul26,

what do you mean by marked?

**Reply by tomboul26** [reply]
> what do you mean by marked?

when you put a material on a component with the small paint pot

with the extension "helper attribute" I could see that some component has the dictionary "ladb_opencutlist" and its methods and properties
I saw for example that in "ladb_opencutlist" materials, the attribute "type" contains a numeric value from 0 to 5, (0: no type, 1: ?, 2: ?, 3 = bar, 4?, 5: tools)
in my sketchup project, i use only 0, 3, or 5
